mono/mini/main.c

```diff
@@ -179,8 +182,49 @@ probe_embedded (const char *program, int *ref_argc, char **ref_argv [])
 		goto doclose;
 	if (read (fd, sigbuffer, sizeof (sigbuffer)) == -1)
 		goto doclose;
-	if (memcmp (sigbuffer+sizeof(uint64_t), "xmonkeysloveplay", 16) != 0)
+	if (memcmp (sigbuffer+sizeof(uint64_t), "xmonkeysloveplay", 16) == 0)
+		goto found;
+	else
+	{
+#ifdef HOST_DARWIN
+		struct mach_header_64 h;
+		if ((sigstart = lseek (fd, 0, SEEK_SET)) == -1)
+			goto doclose;
+		if (read (fd, &h, sizeof (h)) == -1)
+			goto doclose;
+		if (h.magic != MH_MAGIC_64)
+			goto doclose;
+
+		off_t total = h.sizeofcmds;
```

Comment:
A short explanation of what's going on in the code following here would be nice, since I believe a bunch of the stuff in this block is not documented by apple. It mostly makes sense to me, but I'm still hesitant to merge this without some sort of comment explaining it.

Reply:
I have added a few blocks of comments for clarity in https://github.com/mono/mono/pull/19580/commits/343151120e5940871febfda4b70df3b58df3af01